spec/integration/type/file_spec.rb

```diff
@@ -1528,4 +1528,161 @@ def pretty_transaction_error(transaction)
 
     raise "Got #{status_failures.length} failure(s) while applying: #{status_fail_msg}"
   end
+
+  describe "copying a file that is a link to a file" do
+    let(:target) { tmpfile('target') }
+    let(:link) { tmpfile('link') }
+    let(:copy) { tmpfile('copy') }
+    it "should copy the target of the link if :links => follow" do
+      catalog.add_resource described_class.new(
+        :name => target,
+        :ensure => "present",
+        :content => "Jenny I got your number / I need to make you mine")
+      catalog.add_resource described_class.new(
+        :name => link,
+        :ensure => "link",
+        :target => target)
+      catalog.add_resource described_class.new(
+        :name => copy,
+        :ensure => "present",
+        :source => link,
+        :links => "follow")
+      catalog.apply
+      expect(File.ftype(target)).to eq(File.ftype(copy))
```

Comment:
If I guard for `Puppet.features.manages_symlinks?`, is it still necessary to use `Puppet::Filesystem.file?` or `Puppet::FileSystem.symlink?`? That is, if Windows _does_ support symlinks, does `File.ftype()` still do the wrong thing?